src/Tribe/RSVP.php

```diff
@@ -219,6 +219,155 @@ public function hooks() {
+		$template = tribe( 'tickets.editor.template' );
+
+		$ticket = $rsvp->get_ticket( $post_id, $ticket_id );
+
+		// No ticket found.
+		if ( null === $ticket ) {
+			return '';
+		}
+
+		// Set required template globals.
+		$args = [
+			'rsvp_id'    => $ticket_id,
+			'post_id'    => $post_id,
+			'rsvp'       => $ticket,
+			'step'       => $step,
+			'must_login' => ! is_user_logged_in() && $rsvp->login_required(),
+			'login_url'  => Tribe__Tickets__Tickets::get_login_url( $post_id ),
+			'threshold'  => $blocks_rsvp->get_threshold( $post_id ),
+		];
+
+		$args['process_result'] = $this->process_rsvp_step( $args );
+
+		/**
+		 * Allow filtering of the template arguments used.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $args {
+		 *      The list of step template arguments.
+		 *
+		 *      @type int                           $rsvp_id    The RSVP ticket ID.
+		 *      @type int                           $post_id    The ticket ID.
+		 *      @type Tribe__Tickets__Ticket_Object $rsvp       The RSVP ticket object.
+		 *      @type null|string                   $step       Which step being rendered.
+		 *      @type boolean                       $must_login Whether login is required to register.
+		 *      @type string                        $login_url  The site login URL.
+		 *      @type int                           $threshold  The RSVP ticket threshold.
+		 * }
+		 */
+		$args = apply_filters( 'tribe_tickets_rsvp_render_step_template_args', $args );
+
+		// Add the rendering attributes into global context.
+		$template->add_template_globals( $args );
+
+		$html  = $template->template( 'v2/components/loader/loader', [], false );
+		$html .= $template->template( 'v2/rsvp/content', $args, false );
+
+		$response['html'] = $html;
+
+		return wp_send_json_success( $response );
+	}
+
+	/**
+	 * Handle processing the RSVP step based on current arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $args {
+	 *      The list of step template arguments.
+	 *
+	 *      @type int                           $rsvp_id    The RSVP ticket ID.
+	 *      @type int                           $post_id    The ticket ID.
+	 *      @type Tribe__Tickets__Ticket_Object $rsvp       The RSVP ticket object.
+	 *      @type null|string                   $step       Which step being rendered.
+	 *      @type boolean                       $must_login Whether login is required to register.
+	 *      @type string                        $login_url  The site login URL.
+	 *      @type int                           $threshold  The RSVP ticket threshold.
+	 * }
+	 *
+	 * @return array The process result.
+	 */
+	public function process_rsvp_step( array $args ) {
+		$result = [
+			'success' => true,
+			'errors'  => [],
+		];
+
+		// Process the attendee.
+		if ( 'success' === $args['step'] ) {
+			$this->generate_tickets( $args['post_id'] );
+		} elseif ( 'optin' === $args['step'] ) {
+			// @todo Optin for each attendee in order.
+
+			/*$optout = true;
```

Comment:
* :warning: This comment is 57% valid code; is this commented out code?
* :no_entry_sign: Block comment text must start on a new line

<sub>hash: 2e390bc1ba22f38378126e15b284d7d7</sub>